src/app/Dashboard/components/DashboardContainer.js

```diff
@@ -65,36 +81,105 @@ export class DashboardContainer extends Component {
     this.setState({ user: userInfo });
     fetchUserActivites(userInfo.id);
     loadCategories();
+    document.addEventListener('mousedown', this.hideFilter, false);
   }
 
   logPointsModal = () => {
     const { logPoints } = this.state;
     this.setState({
       logPoints: !logPoints,
     });
-  }
+  };
+
+  handleClick = index => (event) => {
+    if (event.target.value === 'Select All') {
+      const filterBy = this.state.filterBy.map((item) => {
+        if (event.target.checked) {
+          return (item.checked = true), item;
+        }
+
+        return (item.checked = false), item;
+      });
+
+      this.setState({ filterBy, filteredUserActivities: null });
+    } else {
+      const filterBy = this.state.filterBy.slice();
+      filterBy[index].checked = !filterBy[index].checked;
+
+      if (!filterBy[index].checked) {
+        const filteredUserActivities = (
+          this.state.filteredUserActivities || this.props.userActivities
```

Comment:
@Kachulio1 It would be nice if you destructured the state and props

Reply:
Yes, I wanted to but the name would conflict with the variable on line 110

Reply:
and also  `this.state.` acts like namespace, it helps with readability